exporters/otlp/include/opentelemetry/exporters/otlp/otlp_http_client.h

```diff
@@ -98,14 +104,16 @@ class OtlpHttpClient
    * Create an OtlpHttpClient using the given options.
    */
   explicit OtlpHttpClient(OtlpHttpClientOptions &&options);
+#ifdef ENABLE_TEST
+  VIRTUAL_TEST ~OtlpHttpClient() {}
```

Comment:
I am thinking if mocking `ext::http::client::HttpClient` would be a better approach (the way we do it for Zipkin) as this will allow us to test most of the existing code in OtlpHttpClient::Export() method. We may have to add ext::http::client::HttpClient object as dependency injection to OtlpHttpClient to make this feasible.  What do you think?

Reply:
I didn't use template because I thought it will be too much of noise in code.
Thanks for the idea, I will use dependency injection.